packages/gatsby-theme-learning/src/hooks/use-update-user.ts

```diff
@@ -0,0 +1,93 @@
+import { useState, useContext, useCallback } from 'react';
+import ConfigContext from '../components/config-context';
+import type { QuizAttempt } from '../components/quiz';
+import { useAuthToken } from './use-auth-token';
+import { User } from '@auth0/auth0-react';
+import { MaintenanceModeError, ServiceDownError } from './use-attempt';
+
+type UpdateUserParams = {
+  userId: string;
+};
+
+export const useUpdateUser = (updateUserParams: UpdateUserParams) => {
+  const { learnApiBaseUrl } = useContext(ConfigContext);
+  const { userId } = updateUserParams;
+  const { getAuthToken } = useAuthToken();
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | undefined>();
+  const [data, setData] = useState<QuizAttempt | undefined>();
+  const [correlationId, setCorrelationId] = useState<string | undefined>();
+
+  const updateUser = useCallback(
+    async (newUser: User): Promise<Response> => {
+      const apiEndpoint = `${learnApiBaseUrl}/api/users/${userId}`;
+      const accessToken = await getAuthToken();
+      const data = await fetch(apiEndpoint, {
+        method: 'POST',
+        body: JSON.stringify(newUser),
```

Comment:
I think you need to add the `credentials` property here. ([docs](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#sending_a_request_with_credentials_included))

It is responsible for sending the `user_session` cookie along with the request, without it we will be creating a new learn-api JWT with every request.

Its a bit tricky as the possible values are `include`, `same-origin` or `omit`
I guess we can make it `credentials: 'include'` on testing only. which will include the cookie cross-domain
but make it; `same-origin` on production. because at that point `docs.commercetools.com` and `api.learn.commercetools.com` are same domain 

WDYT?

Reply:
I implemented what you suggested, maybe you can have a check and see if it's working as expected

Reply:
P.S:: from a security standpoint "include" is okay. The actual locking down happens in the CORS headers that the API server sends and that has an allowlist for our preview and production domains.